src/Pos/Util/UserSecret.hs

```diff
@@ -109,6 +158,7 @@ peekUserSecret path = takeReadLock path $ do
 -- 'writeUserSecretRelease'.
 takeUserSecret :: (MonadIO m) => FilePath -> m UserSecret
 takeUserSecret path = liftIO $ do
+    initializeUserSecret path
```

Comment:
If it's a shared directory (like /tmp without per-user tmp), then you defintely need to keep your fd after creation (specially relevant for tmp file). otherwise for a user directory, that is presumably not readable/writable by anyone else that probably doesn't matter much